sdks/java/core/src/main/java/org/apache/beam/sdk/util/MergingActiveWindowSet.java

```diff
@@ -448,6 +448,11 @@ public W mergedWriteStateAddress(Collection<W> toBeMerged, W mergeResult) {
     return mergeResult;
   }
 
+  @Override
+  public boolean hasState() {
+    return !activeWindowToStateAddressWindows.isEmpty();
```

Comment:
Is the intention to answer the query for committed state, or hypothetically if the current transient state were to be committed? This is consistent with the latter. If the former, you'd want `valueState.hasState`. Then this would be `ReadableState<Boolean>` as well.